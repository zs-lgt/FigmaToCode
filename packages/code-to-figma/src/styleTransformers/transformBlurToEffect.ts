import { colorToRGBA } from './transformColors';
import { BlurProperties, BlendStyleProperties } from './transformBlendProperties';

export const transformBlurToEffect = (styles: Partial<BlendStyleProperties>): BlurEffect[] => {
    let blurs: Partial<BlurProperties>[] = [styles];
    if ('blurs' in styles && Array.isArray(styles.shadows)) {
        blurs = styles.blurs;
    }
    return blurs.map(blur => ({
        type: blur.blurType != null ? blur.blurType : 'LAYER_BLUR',
        radius: blur.blurRadius || 0,
        visible: true
    }));
};
