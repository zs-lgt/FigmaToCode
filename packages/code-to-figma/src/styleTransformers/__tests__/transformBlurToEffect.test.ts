import { transformBlurToEffect } from '../transformBlurToEffect';

describe('transformBlurToEffect', () => {
    it('blur radius', () => {
        const result = transformBlurToEffect({
            blurRadius: 4
        });
        expect(result).toMatchSnapshot();
    });
});
