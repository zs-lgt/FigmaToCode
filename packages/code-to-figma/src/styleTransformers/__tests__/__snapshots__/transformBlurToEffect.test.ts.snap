// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformBlurToEffect blur radius 1`] = `
Array [
  Object {
    "radius": 4,
    "type": "LAYER_BLUR",
    "visible": true,
  },
]
`;
